src/scanner.ll

```diff
@@ -153,10 +153,17 @@
 [0-9]+                           { try {
                                      return yy::parser::make_NUMBER(std::stoi(yytext, NULL, 10), yylloc);  
                                    } catch (...) { 
-                                     driver.error(yylloc, "integer constant must be in range [0, 2147483647]");
+                                     driver.error(yylloc, "positive integer constant must be in range [0, 2147483647]");
```

Comment:
s/positive/non-negative/

QQ since you are changing the scanner. Is it worth changing the regular expression for decimals to 0|([1-9][0-9]*) to prevent a number in octal notation from being interpreted as a decimal number?